specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/hyperdrive.json

```diff
@@ -0,0 +1,714 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "HyperDrive REST API",
+    "version": "2019-05-01",
+    "title": "HyperDrive"
+  },
+  "tags": [],
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/hyperdrive/v1.0/{arm_scope}/runs": {
+      "post": {
+        "tags": [
+          "HyperDriveExperiment"
+        ],
+        "summary": "Create an Experiment.",
+        "description": "Create a HyperDrive Experiment.",
+        "operationId": "HyperDrive_CreateExperiment",
+        "consumes": [
+          "multipart/form-data"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ArmScopeParameter"
+          },
+          {
+            "in": "formData",
+            "name": "config",
+            "description": "The configuration file with experiment JSON content. A text file that is a JSON-serialized '#/definitions/HyperDriveCreateExperiment' object.",
+            "type": "file",
+            "required": true
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/HyperDriveExperimentResponse"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/HyperDriveErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create a HyperDrive experiment": {
+            "$ref": "./examples/createHyperDriveExperiment.json"
+          }
+        }
+      }
+    },
+    "/hyperdrive/v1.0/{arm_scope}/runs/{run_id}/cancel": {
+      "post": {
+        "tags": [
+          "HyperDriveExperiment"
+        ],
+        "summary": "Cancel an Experiment.",
+        "description": "Cancel a HyperDrive Experiment.",
+        "operationId": "HyperDrive_CancelExperiment",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ArmScopeParameter"
+          },
+          {
+            "name": "run_id",
+            "in": "path",
+            "description": "Hyperdrive run id to cancel.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "header",
+            "name": "RunHistoryHost",
```

Comment:
How can a client determine what values they can provide here?

Reply:
And what happens if the value isn't specified?

Reply:
Already removed 'required' true. HD run should get cancelled without RunHistoryHost.

Reply:
This can be resolved.

Reply:
No further action.